src/test/java/teammates/test/cases/common/TimeHelperTest.java

```diff
@@ -86,6 +88,158 @@ public void testCombineDateTime() throws ParseException{
     }
     
     @Test
+    public void testIsTimeWithinPeriod() {
+        Calendar startCalendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        Calendar endCalendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        Calendar timeCalendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        
+        // Set start time to 5 days before today and end time to 5 days after today
+        startCalendar.add(Calendar.DAY_OF_MONTH, -5);
+        endCalendar.add(Calendar.DAY_OF_MONTH, 5);
+        
+        Date startTime = startCalendar.getTime();
+        Date endTime = endCalendar.getTime();
+        Date time;
+        
+        
+        
+        ______TS("Time within period test");
+        time = timeCalendar.getTime();
+        
+        ______TS("Time within period: include start and end time");
+        assertTrue(TimeHelper.isTimeWithinPeriod(startTime, endTime, time, true, true));
+
+        ______TS("Time within period: include start and exclude end time");
+        assertTrue(TimeHelper.isTimeWithinPeriod(startTime, endTime, time, true, false));
+
+        ______TS("Time within period: exclude start and include end time");
+        assertTrue(TimeHelper.isTimeWithinPeriod(startTime, endTime, time, false, true));
+
+        ______TS("Time within period: exclude start and end time");
+        assertTrue(TimeHelper.isTimeWithinPeriod(startTime, endTime, time, false, false));
```

Comment:
Personally, I don't think you need to spell out every single detail of each test cases. This should suffice:
```java
Date time;

______TS("Time within period test");
time = timeCalendar.getTime();

assertTrue(...);
assertTrue(...);
assertTrue(...);
assertTrue(...);

______TS("Time on start time test");
...
```